go-controller/pkg/util/go_ovn.go

```diff
@@ -0,0 +1,543 @@
+package util
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"syscall"
+
+	goovn "github.com/ebay/go-ovn"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util/ovnbindings"
+	"k8s.io/klog"
+)
+
+type OVNClient struct {
+	client goovn.Client
+}
+
+var _ ovnbindings.OVNInterface = &OVNClient{}
+
+func NewOVNNBClient() (ovnbindings.OVNInterface, error) {
+	var (
+		err         error
+		ovnNBClient *OVNClient
+	)
+
+	ovnNBClient = &OVNClient{}
+
+	switch config.OvnNorth.Scheme {
+	case config.OvnDBSchemeSSL:
+		ovnNBClient.client, err = initGoOvnSslClient(config.OvnNorth.Cert,
+			config.OvnNorth.PrivKey, config.OvnNorth.CACert,
+			config.OvnNorth.GetURL(), goovn.DBNB, config.OvnNorth.CertCommonName)
+	case config.OvnDBSchemeTCP:
+		ovnNBClient.client, err = initGoOvnTcpClient(config.OvnNorth.GetURL(), goovn.DBNB)
+	case config.OvnDBSchemeUnix:
+		ovnNBClient.client, err = initGoOvnUnixClient(config.OvnNorth.GetURL(), goovn.DBNB)
+	default:
+		err = fmt.Errorf("invalid db scheme: %s when initializing the OVN NB Client",
+			config.OvnNorth.Scheme)
+	}
+
+	if err != nil {
+		return nil, fmt.Errorf("couldn't initialize NBDB client: %s", err)
+	}
+
+	klog.Infof("Created OVN NB client with Scheme: %s", config.OvnNorth.Scheme)
+	return ovnNBClient, nil
+}
+
+func NewOVNSBClient() (ovnbindings.OVNInterface, error) {
+	var (
```

Comment:
Is there going to be ever a need for more than one instance of ovnSBClient needed? in other words, would a singleton suffice?

Reply:
You could if we ever decide to have more than one controller and sub-divide processing. So theoretically yes.